### PR TITLE
Window app rework

### PR DESCRIPTION
**Pull request by andygallay-silabs**

### CODE REVIEW

File: examples/window-app/common/src/WindowApp.cpp [hunk 1]
```diff
@@ -345,6 +345,7 @@ void WindowApp::DispatchEventAttributeChange(chip::EndpointId endpoint, chip::At
     default:
         break;
     }
+
```
Comment (jepenven-silabs) [inline examples/window-app/common/src/WindowApp.cpp]: Should not be committed

File: examples/window-app/silabs/efr32/args.gni [hunk 2]
```diff
@@ -24,4 +24,4 @@ chip_enable_openthread = true
 openthread_external_platform =
     "${chip_root}/third_party/openthread/platforms/efr32:libopenthread-efr32"
 
-use_base_app = false
+use_base_app = true
```
Comment (jepenven-silabs) [inline examples/window-app/silabs/efr32/args.gni]: This build option should be removed at large since it's not needed anymore

File: examples/window-app/silabs/efr32/build_for_wifi_args.gni [hunk 3]
```diff
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import("//build_overrides/chip.gni")
+import("${chip_root}/config/standalone/args.gni")
```
Comment (jepenven-silabs) [inline examples/window-app/silabs/efr32/build_for_wifi_args.gni]: Why is this needed?

File: examples/window-app/silabs/efr32/include/WindowManager.h [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+    static WindowManager sInstance;
+    struct Timer
+    {
+        typedef void (*Callback)(Timer & timer);
+
+        Timer(const char * name, uint32_t timeoutInMs, Callback callback, void * context);
+
+        void Start();
+        void Stop();
```
Comment (jepenven-silabs) [inline examples/window-app/silabs/efr32/include/WindowManager.h]: Why would we want to stop the Window manager? If unused please remove

File: examples/window-app/silabs/efr32/include/WindowManager.h [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+#endif
+
+using namespace chip::app::Clusters::WindowCovering;
+
+class WindowManager
+{
+public:
+
+    static WindowManager sInstance;
+    struct Timer
```
Comment (jepenven-silabs) [inline examples/window-app/silabs/efr32/include/WindowManager.h]: Is this whole struct needed?

File: examples/window-app/silabs/efr32/include/WindowManager.h [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+        bool mIsActive        = false;
+
+        TimerHandle_t mHandler = nullptr;
+        void IsrStart();
+
+    private:
+        static void TimerCallback(TimerHandle_t xTimer);
+    };
+
+    struct Button
```
Comment (jepenven-silabs) [inline examples/window-app/silabs/efr32/include/WindowManager.h]: Is this struct really needed?

File: examples/window-app/silabs/efr32/include/WindowManager.h [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+        static void CallbackOperationalStateSet(intptr_t arg);
+    };
+
+    static WindowManager & Instance();
+
+    WindowManager();
+
+    CHIP_ERROR Init();
+    CHIP_ERROR Start();
+    CHIP_ERROR Run();
```
Comment (jepenven-silabs) [inline examples/window-app/silabs/efr32/include/WindowManager.h]: Should cleanup all of this with the Start, Run, Stop and finish

File: scripts/pigweed.json [hunk 8]
```diff
@@ -0,0 +1,134 @@
+{
+  "packages": [
+    {
+      "path": "gn/gn/${platform}",
+      "platforms": [
```
Comment (jepenven-silabs) [inline scripts/pigweed.json]: This file should not be committed

**Review by jepenven-silabs — CHANGES_REQUESTED** [review]
Rework needed. Seems like the initial implementation was simply wrapped inside an Apptask without any cleanup whatsoever. Please cleanup the Window Manager and make it more streamlined with other examples

File: examples/window-app/silabs/efr32/args.gni (hunk 2 above)
Comment (andygallay-silabs) [reply examples/window-app/silabs/efr32/args.gni]: Removed. But so far, I need to specify use_base_app = true when building the application through the terminal:

`./scripts/examples/gn_efr32_example.sh ./examples/window-app/silabs/efr32/ ./out/window-app BRD4187C use_base_app=true`

File: examples/window-app/silabs/efr32/build_for_wifi_args.gni (hunk 3 above)
Comment (andygallay-silabs) [reply examples/window-app/silabs/efr32/build_for_wifi_args.gni]: A tentative to integrate the common window app data model to the silabs folder.

File: examples/window-app/silabs/efr32/include/WindowManager.h (hunk 5 above)
Comment (andygallay-silabs) [reply examples/window-app/silabs/efr32/include/WindowManager.h]: It's the Timer struct from WindowApp (base class) and WindowAppImpl combined, since WindowAppImpl extended the definition of the base class Timer struct.

File: examples/window-app/silabs/efr32/include/WindowManager.h (hunk 4 above)
Comment (andygallay-silabs) [reply examples/window-app/silabs/efr32/include/WindowManager.h]: It's the Timer stop function, used in line 452 and 516, 955, 989 in WindowManager.cpp.

File: examples/window-app/silabs/efr32/include/WindowManager.h (hunk 6 above)
Comment (andygallay-silabs) [reply examples/window-app/silabs/efr32/include/WindowManager.h]: The struct is used within the WindowManager to define the actions of the buttons on the board. Maybe the buttons behavior could be called in AppTask, like the lightning app example ?

File: examples/window-app/silabs/efr32/include/WindowManager.h (hunk 7 above)
Comment (andygallay-silabs) [reply examples/window-app/silabs/efr32/include/WindowManager.h]: Will merge Init and Start into a single Init function. Removed Finish function since we don't care about memory leaks in our use case. I can merge the code within WindowManager::Run() into the WindowManager::OnTaskCallback where the function Run() is called.

File: examples/window-app/silabs/efr32/args.gni (hunk 2 above)
Comment (jepenven-silabs) [reply examples/window-app/silabs/efr32/args.gni]: This should be removed at large, meaning removed everywhere, even beyond the window-app...

File: examples/window-app/silabs/efr32/include/AppEvent.h [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        kEventType_WinkOn,
+    };
+
+    uint16_t Type;
+    chip::EndpointId mEndpoint = 0;
+    chip::AttributeId mAttributeId;
+    
+    union
+    {
+        struct
```
Comment (jepenven-silabs) [inline examples/window-app/silabs/efr32/include/AppEvent.h]: Are all these struct needed

File: examples/window-app/silabs/efr32/include/AppTask.h [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+#include <ble/BLEEndPoint.h>
+#include <lib/core/CHIPError.h>
+#include <platform/CHIPDeviceLayer.h>
+
+/**********************************************************
+ * Defines
+ *********************************************************/
+
+// Application-defined error codes in the CHIP_ERROR space.
+#define APP_ERROR_EVENT_QUEUE_FAILED CHIP_APPLICATION_ERROR(0x01)
```
Comment (jepenven-silabs) [inline examples/window-app/silabs/efr32/include/AppTask.h]: These defines are probably not needed please removed

File: examples/window-app/silabs/efr32/include/AppTask.h [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+    static void AppTaskMain(void * pvParameter);
+
+    CHIP_ERROR StartAppTask();
+
+    /**
+     * @brief Callback called by the identify-server when an identify command is received
+     *
+     * @param identify identify structure the command applies on
+     */
+    static void OnIdentifyStart(Identify * identify);
```
Comment (jepenven-silabs) [inline examples/window-app/silabs/efr32/include/AppTask.h]: this will have to go after the rebase

File: examples/window-app/silabs/efr32/include/WindowManager.h (hunk 5 above)
Comment (jepenven-silabs) [reply examples/window-app/silabs/efr32/include/WindowManager.h]: That doesn't answer the question. Is is really needed ? couldn't we just use the FreeRTOS api directly ? if so please removed.

File: examples/window-app/silabs/efr32/include/WindowManager.h (hunk 4 above)
Comment (jepenven-silabs) [reply examples/window-app/silabs/efr32/include/WindowManager.h]: Then couldn't we just use the FreeRTOS api directly instead of going through a wrapper ?

File: examples/window-app/silabs/efr32/include/WindowManager.h [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+    CHIP_ERROR Init();
+    CHIP_ERROR UpdateState();
+    void PostAttributeChange(chip::EndpointId endpoint, chip::AttributeId attributeId);
+
+    static void ButtonEventHandler(uint8_t button, uint8_t btnAction);
+    void OnButtonChange(uint8_t button, uint8_t btnAction);
+
+    Timer * CreateTimer(const char * name, uint32_t timeoutInMs, WindowManager::Timer::Callback callback,
+                                   void * context);
+    Button * CreateButton(WindowManager::Button::Id id, const char * name);
```
Comment (jepenven-silabs) [inline examples/window-app/silabs/efr32/include/WindowManager.h]: Is this really needed ?

File: examples/window-app/silabs/efr32/src/AppTask.cpp [hunk 13]
```diff
@@ -0,0 +1,251 @@
...
+#include <platform/silabs/platformAbstraction/SilabsPlatform.h>
+
+#include <setup_payload/QRCodeSetupPayloadGenerator.h>
+#include <setup_payload/SetupPayload.h>
+
+#include <lib/support/CodeUtils.h>
+
+#include <platform/CHIPDeviceLayer.h>
+
+#if defined(SL_CATALOG_SIMPLE_LED_LED1_PRESENT)
```
Comment (jepenven-silabs) [inline examples/window-app/silabs/efr32/src/AppTask.cpp]: This should be removed for 917 compatibility

File: examples/window-app/silabs/efr32/src/AppTask.cpp [hunk 14]
```diff
@@ -0,0 +1,251 @@
...
+    ChipLogProgress(Zcl, "Trigger Identify Complete");
+    sIdentifyEffect = EMBER_ZCL_IDENTIFY_EFFECT_IDENTIFIER_STOP_EFFECT;
+
+#if CHIP_DEVICE_CONFIG_ENABLE_SED == 1
+    AppTask::GetAppTask().StopStatusLEDTimer();
+#endif
+}
+} // namespace
+
+void OnTriggerIdentifyEffect(Identify * identify)
```
Comment (jepenven-silabs) [inline examples/window-app/silabs/efr32/src/AppTask.cpp]: This will have to be removed after the rebase

File: examples/window-app/silabs/efr32/include/AppEvent.h (hunk 9 above)
Comment (andygallay-silabs) [reply examples/window-app/silabs/efr32/include/AppEvent.h]: Removed LightEvent
